e2eshark/onnx/models/opt-125M-awq/model.py

```diff
@@ -16,34 +16,92 @@
 # to form runmodel.py in the rundirectory which is then taken
 # through flow
 
-# set some variable parameters (dynamic dims)
-batch_size = 3 
-sequence_length = 7 
-past_sequence_length = 0 # setting this parameter to anything other than 0 results in an error
+# If you want to run the test with a static model.onnx, set run_as_static = True
+run_as_static = False
+dim_params = (
+    "batch_size",
+    "sequence_length",
+    "past_sequence_length",
+    "past_sequence_length + 1",
+)
+dim_values = (12, 7, 0, 1)
+pv_zip = zip(dim_params, dim_values)
+pv = dict(pv_zip)
+
+model = onnx.load("model.onnx")
+isDynamic = bool(model.graph.input[0].type.tensor_type.shape.dim[0].dim_param)
+
+if run_as_static and isDynamic:
+    print("Saving dynamic model...")
+    onnx.save(model, "dynamic_model.onnx")
+    print("\t dynamic model saved as dynamic_model.onnx")
```

Comment:
> I'm not sure how I feel about saving the dynamic model to the test bin. Any suggestions?

One thing we've been setting up in [`iree_tests/`](https://github.com/nod-ai/SHARK-TestSuite/tree/main/iree_tests) is an external cache directory that you can set via an environment variable (or CLI flag). For individual developers, a directory structure under the repository itself is usually fine (as long as it is clearly marked, added to `.gitignore`, and is easy to clean). For CI jobs and on shared machines, prefer to store and reference large files that don't change test run to test run in an external cache directory.

Reply:
Hm, I'll need to look at how that is set up. This approach, in general, might be more annoying from a CI perspective than just adding a separate test entirely for a static model.